src/utils/composables/useFieldValidation.js

```diff
@@ -0,0 +1,132 @@
+import { computed, onMounted, ref, toRef, watch } from 'vue'
+
+export function propsValidator4ValidateFunc(value, _props) {
+    if (value === null) {
+        return true
+    }
+    const returnObject = value('')
+    if (typeof returnObject !== 'object') {
+        return false
+    }
+    if ((!'valid') in returnObject) {
+        return false
+    }
+    if ((!'invalidMessage') in returnObject) {
+        return false
+    }
+    if (typeof returnObject.valid !== 'boolean') {
+        return false
+    }
+    if (
+        typeof returnObject.invalidMessage !== 'string' &&
+        !(returnObject.invalidMessage instanceof 'String')
+    ) {
+        return false
+    }
+    return true
+}
+
+export function useFieldValidation(
+    props,
+    model,
+    emits,
+    {
+        customValidate = () => {
+            return { valid: true, invalidMessage: '' }
+        },
+        requiredInvalidMessage = 'field_required',
+    } = {}
+) {
```

Comment:
Could you add a bit of documentation that describe what inputs are expected and what they do?

Reply:
Done